### ISSUE TITLE
Search results are not accurate when nprobe >= nlist

### ISSUE THREAD

**Issue by BossZou** [milvus-io/milvus#2376]
**Describe the bug**

I run ann_benchmark  test procedure, finding the search results are not accurate when nprobe >= nlist.

--------
Index type: IVF_FLAT

args:
```
args: [['IVF_FLAT'], [128]]  # nlist
          query-args: [[128, 200]]  # nprobe (should be <= nlist)
```

--------
The dataset is [sift-128-euclidean](http://ann-benchmarks.com/%s.hdf5).

--------
ann_benchmark test repo: https://github.com/BossZou/ann-benchmarks/tree/ann-update-090

**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**

**Environment details**
- Hardware/Softward conditions (OS, CPU, GPU, Memory)
- Method of installation: Docker
- Milvus version: v0.9.0
- Milvus configuration: see [config_file](https://raw.githubusercontent.com/milvus-io/milvus/master/core/conf/demo/server_config.yaml)

**Screenshots**

**Additional context**

**Comment by BossZou** [comment]
This is because that milvus vill automatically  adjust param `nlist` if data count is small than 1 million.

**Comment by shengjun1985** [comment]
It is fixed in #2399.
